common/network_manager.go

```diff
@@ -57,6 +57,10 @@ func (c *Credentials) Parse(ctx context.Context, credentials sobek.Value) error
 	return nil
 }
 
+type metricInterceptor interface {
+	urlTagName(ctx context.Context, urlTag string) (string, bool)
+}
```

Comment:
It's fine this way. But why don't we use a simple function value instead of declaring an interface—if this is only about testing? I wouldn't use an interface only for testing as they are mostly for extending the functionality for unforeseen cases (think io.Writer).

Reply:
Right, that didn't occur to me earlier. I think i'll leave it as an interface though as it's not critical.

Reply:
Sure.